app/src/main/java/org/mozilla/fenix/components/toolbar/BrowserToolbarView.kt

```diff
@@ -227,7 +227,23 @@ class BrowserToolbarView(
 
     fun expand() {
         if (!settings.shouldUseBottomToolbar) {
-            layout.app_bar?.setExpanded(true)
+            view.app_bar?.setExpanded(true)
+        }
+    }
+
+    /**
+     * Performs all actions that refer to the BrowserToolbarView that don't require user action.
+     * They are set in the callback because we need to ensure phones execute the code before
+     * we are done inflating.
+     */
+    fun loadAsync(action: View.() -> Unit) {
+        AsyncLayoutInflater(container.context).inflate(toolbarLayout, container) { finalView, _, parent ->
+            view = finalView.findViewById((R.id.toolbar))
+            with(parent!!) {
+                addView(finalView)
+                setupView()
+                action()
```

Comment:
I think we should pass the `finalView` as a parameter to the action lambda. Basically, the same as Mike suggested here: https://github.com/mozilla-mobile/fenix/pull/9003/files#r391224957